ngmix/pixels.py

```diff
@@ -1,18 +1,51 @@
 import numpy
+from .gexceptions import GMixFatalError
 
-def make_pixels(image, weight, jacob):
+def make_pixels(image, weight, jacob, ignore_zero_weight=True):
     """
     make a pixel array from the image and weight
+
+    stores v,u image value, and 1/err for each pixel
+
+
+    parameters
+    ----------
+    pixels: array
+        1-d array of pixel structures, u,v,val,ierr
+    image: 2-d array
+        2-d image array
+    weight: 2-d array
+        2-d image array same shape as image
+    jacob: jacobian structure
+        row0,col0,dvdrow,dvdcol,dudrow,dudcol,...
+
+    ignore_zero_weight: bool
+        If set, zero or negative weight pixels are ignored.  In this case the
+        returned pixels array is equal in length to the set of positive weight
+        pixels in the weight image.  Default True.
+
+    returns
+    -------
+    1-d pixels array
     """
     from .pixels_nb import fill_pixels
 
-    pixels = numpy.zeros(image.size, dtype=_pixels_dtype)
+    if ignore_zero_weight:
+        w=numpy.where(weight > 0.0)
+        if w[0].size == 0:
+            raise GMixFatalError('no weights > 0')
```

Comment:
Do you want a more specific subclass Error here so that people can distinguish what went wrong?

Reply:
I could not think of a sensible thing for this other than fatal